controllers/custom-forms.ts

```diff
@@ -63,3 +66,87 @@ customFormRouter.get('/find-usage', async (req: Request, res: Response) => {
     users,
   });
 });
+
+customFormRouter.post(
+  '/field/is-uri-allowed-as-path',
+  async (req: Request, res: Response) => {
+    const uri = req.body?.uri?.trim() ?? null;
+    if (!uri) {
+      throw new HttpError('No uri was provided.', 400);
+    }
+
+    const regex = '^(https?://)[a-zA-Z0-9-]+(.[a-zA-Z0-9-]+)*(/.*)?$';
+    const uriRegex = new RegExp(regex);
+    const isValid = uriRegex.test(uri);
+    const hasSpaces = /\s/.test(uri);
+
+    const illegalUris = [
+      RDF('type'),
+      MU('uuid'),
+      DCTERMS('modified'),
+      DCTERMS('isVersionOf'),
+      FORM('displayType'),
+      FORM('includes'),
+      FORM('initGenerator'),
+      FORM('validatedBy'),
+      FORM('targetType'),
+      FORM('targetLabel'),
+      FORM('initGenerator'),
+      FORM('prototype'),
+      FORM('dataGenerator'),
+      FORM('shape'),
+      FORM('prefix'),
+      FORM('forType'),
+      FORM('showInSummary'),
+      SHACL('name'),
+      SHACL('order'),
+      SHACL('datatype'),
+      SHACL('path'),
+      SHACL('group'),
+      SHACL('severity'),
+      SHACL('resultMessage'),
+      EXT('ValueToCompare'),
+      EXT('prefix'),
+      EXT('withHistory'),
+      EXT('isLibraryEntryField'),
+      EXT('hasUserInputPath'),
+      LMB('hasPublicationStatus'),
+      MANDAAT('bekrachtigtAanstellingVan'),
+    ];
+    const isAllowed = !illegalUris.includes(uri);
+
+    const formId = req.body?.formId ?? null;
+    const fieldUri = req.body?.fieldUri ?? null;
+    let isUniquePathInForm = true;
+    if (formId && isValid) {
+      isUniquePathInForm = await isUriUsedAsPredicateInForm(
```

Comment:
sure this is important to check but it actually needs to go further
there shouldn't be instances of the form type that have the uri already (e.g. when extending existing form or a form that may only offer a part of the predicates)